libs/telemetry/src/main/java/org/opensearch/telemetry/metrics/DefaultMetricsRegistry.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.metrics;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+/**
+ * Default implementation for {@link MetricsRegistry}
+ */
+public class DefaultMetricsRegistry implements MetricsRegistry {
+    private final MetricsTelemetry metricsTelemetry;
+
+    /**
+     * Constructor
+     * @param metricsTelemetry metrics telemetry.
+     */
+    public DefaultMetricsRegistry(MetricsTelemetry metricsTelemetry) {
+        this.metricsTelemetry = metricsTelemetry;
+    }
+
+    @Override
+    public Counter createCounter(String name, String description, String unit) {
+        return metricsTelemetry.createCounter(name, description, unit);
+    }
+
+    @Override
+    public Counter createUpDownCounter(String name, String description, String unit) {
+        return metricsTelemetry.createUpDownCounter(name, description, unit);
+    }
+
+    @Override
+    public void close() throws IOException {
+        ((Closeable) metricsTelemetry).close();
```

Comment:
`MetricsTelemetry` is closeable:

```suggestion
metricsTelemetry.close();
```